File: testebackendestagiov2/src/main/java/io/github/pedrobicudo/testebackendestagiov2/model/domain/entities/EquipmentState.java
package io.github.pedrobicudo.testebackendestagiov2.model.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "equipment_state")
public class EquipmentState {

    @Id
    @GeneratedValue
    @Column(updatable = false, nullable = false)
    private UUID id;

    @Column
    private String name;

    @Column
    private String color;

}

File: Equipments.Domain/Entities/EquipmentState.cs


namespace Equipments.Domain.Entities
{
    public class EquipmentState :Entity
    {
        public string Name { get; private set; }
        public string Color { get; private set; }

        public EquipmentState(string name, string color)
        {
            Name = name;
            Color = color;
        }
        public void EditName(string name) => Name = name;
    }
}
